harness/determined/common/experimental/experiment.py

```diff
@@ -149,7 +163,8 @@ def wait(self, interval: float = 5.0) -> ExperimentState:
 
         elapsed_time = 0.0
         while True:
-            exp = bindings.get_GetExperiment(sess, experimentId=self._id).experiment
+            exp = self._get(session=sess)
```

Comment:
should we have the default session for `Determined` client have retries? it'd be annoying to add it for each method individually, or decide which ones are worthy of retries.

Reply:
I asked Ryan this once. He pointed out that some calls aren't idempotent, so without some kind of idempotence token it's dangerous to retry a request that's failed.

Reply:
in the harness we are aggressive and [retry every method](https://github.com/determined-ai/determined/blob/main/harness/determined/common/util.py#L181) (`allowed_methods=False`) which (rarely) blows up for non-idempotent POSTs.

by [default](https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html#urllib3.util.Retry.DEFAULT_ALLOWED_METHODS) `DEFAULT_ALLOWED_METHODS = frozenset({'DELETE', 'GET', 'HEAD', 'OPTIONS', 'PUT', 'TRACE'})`. for the Python SDK we can start off being conservative and reduce this list to `'GET', 'HEAD', 'OPTIONS'` — all of these methods should be idempotent: I suspect some `DELETEs` and `PUTs` may be non-idempotent; we don't use `TRACE`. Python SDK robably does not use `OPTIONS` either, but there's no reason to exclude it.

Reply:
another option might be having the session on the SDK be configurable/public. though i don't know if a user would/should know which methods are able to be retried.

Reply:
implemented default retries in top-level client.

Reply:
#7063